src/sentry/models/organizationmemberinvite.py

```diff
@@ -0,0 +1,142 @@
+import secrets
+from datetime import timedelta
+from enum import Enum
+
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from sentry.backup.dependencies import ImportKind
+from sentry.backup.helpers import ImportFlags
+from sentry.backup.scopes import ImportScope, RelocationScope
+from sentry.db.models import FlexibleForeignKey, region_silo_model, sane_repr
+from sentry.db.models.base import DefaultFieldsModel
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.roles import organization_roles
+
+INVITE_DAYS_VALID = 30
+
+
+class InviteStatus(Enum):
+    APPROVED = 0
+    REQUESTED_TO_BE_INVITED = 1
+    REQUESTED_TO_JOIN = 2
+
+    @classmethod
+    def as_choices(cls):
+        return (
+            (InviteStatus.APPROVED.value, _("Approved")),
+            (
+                InviteStatus.REQUESTED_TO_BE_INVITED.value,
+                _("Organization member requested to invite user"),
+            ),
+            (InviteStatus.REQUESTED_TO_JOIN.value, _("User requested to join organization")),
+        )
+
+
+invite_status_names = {
+    InviteStatus.APPROVED.value: "approved",
+    InviteStatus.REQUESTED_TO_BE_INVITED.value: "requested_to_be_invited",
+    InviteStatus.REQUESTED_TO_JOIN.value: "requested_to_join",
+}
+
+
+def default_expiration():
+    return timezone.now() + timedelta(days=INVITE_DAYS_VALID)
+
+
+def generate_token():
+    return secrets.token_hex(nbytes=32)
+
+
+@region_silo_model
+class OrganizationMemberInvite(DefaultFieldsModel):
+    """
+    Identifies relationships between organizations and their invited users.
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
```

Comment:
Does it make sense to have a relocation scope for this model? I'm under the impression this isn't something we can/should migrate over.